addons/mail/controllers/discuss.py

```diff
@@ -573,3 +573,29 @@ def mail_guest_update_name(self, guest_id, name):
         if guest_to_rename_sudo != guest and not request.env.user._is_admin():
             raise NotFound()
         guest_to_rename_sudo._update_name(name)
+
+    # --------------------------------------------------------------------------
+    # Link preview API
+    # --------------------------------------------------------------------------
+
+    @http.route('/mail/link_preview', methods=['POST'], type='json', auth='public')
+    def mail_link_preview(self, message_id, urls):
+        throttle = request.env['ir.config_parameter'].sudo().get_param('mail.link_preview_throttle', 99)
+        if int(throttle) == 0:
+            return [('clear',)]
+        link_previews = []
+        guest = request.env['mail.guest']._get_guest_from_request(request)
+        for url in set(urls[0:5]):
```

Comment:
why 0:5 specifically?

Reply:
I just realize that this code doesn't do what it's suppose to.
The goal is to create 5 link preview max per message. But this just limit the number of url...